src/mammos_mumag/mesh.py

```diff
@@ -1,8 +1,97 @@
 """Mesh functions."""
 
 import pathlib
+import shutil
 
-_MESH_DIR = pathlib.Path(__file__).parent / "mesh"
+import requests
+import yaml
 
-CUBE_20_nm = _MESH_DIR / "CUBE_20_nm.fly"
-"""Cube with 20 nanometer sides."""
+with open(pathlib.Path(__file__).parent / "mesh" / "README.yaml") as f:
+    DATA = yaml.safe_load(f)
+
+
+def find_mesh(mesh_name: str | None = None) -> list[str]:
+    """Find available meshes matchin given name.
+
+    Args:
+        mesh_name: Desired mesh name. If None, returns all available meshes.
+
+    Returns:
+        List of matches with given name. Empty list if no matches are found.
+    """
+    if mesh_name is None:
+        return list(DATA["meshes"].keys())
+    else:
+        return [n for n in DATA["meshes"] if mesh_name in n]
+
+
+class Mesh:
+    """Mesh class."""
+
+    def __init__(self, mesh_name: str | pathlib.Path):
+        """Initialize Mesh with either mesh_name or path for local meshes."""
+        matches = (
+            find_mesh(mesh_name) if not isinstance(mesh_name, pathlib.Path) else []
+        )
+        if len(matches) > 1:
+            raise ValueError("Mesh name unambiguous. More than one match found.")
+        elif not matches:
+            filepath = pathlib.Path(mesh_name)
+            if filepath.is_file():
+                self.name = filepath
+                self.info = {"description": "User defined mesh."}
+                self._local = True
+            else:
+                raise ValueError("Mesh not found.")
+        else:
+            self.name = matches[0]
+            self.info = DATA["meshes"][matches[0]]
+            self._local = False
+
+    def __str__(self) -> str:
+        """Implement str dunder."""
+        s = f"name:\n\t{self.name}\n"
+        for k, v in self.info.items():
+            s += f"{k}:\n\t{v}\n"
+        return s
+
+    def __repr__(self) -> str:
+        """Implement repr dunder."""
+        return f"Mesh({self.name})"
+
+    def write(self, dest: pathlib.Path | str) -> None:
+        """Write mesh to destination."""
+        dest = pathlib.Path(dest).resolve()
+        if dest.suffix != ".fly":
+            raise ValueError("Only `.fly` meshes are available.")
+        if self._local:
+            shutil.copy(self.name, dest)
+        else:
+            res = requests.get(f"{DATA['metadata']['zenodo_url']}files/{self.name}.fly")
+            if res.status_code != 200:
+                raise ConnectionError("Mesh not found.")
```

Comment:
Not sure if this is the best exception class, it seems to be used for different types of problems (https://docs.python.org/3/library/exceptions.html#ConnectionError)